pkg/controllerhelpers/core_test.go

```diff
@@ -358,3 +423,122 @@ func TestAggregateStatusConditions(t *testing.T) {
+			elems:                      []string{},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   "",
+		},
+		{
+			name:                       "nil slice",
+			elems:                      nil,
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   "",
+		},
+		{
+			name: "single element exceeds limit",
+			elems: []string{
+				strings.Repeat("a", testLimit+1),
+			},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   "(1 more omitted)",
+		},
+		{
+			name: "joined length is less than limit",
+			elems: []string{
+				"a",
+				"b",
+			},
+			separator: testSep,
+			limit:     testLimit,
+			expected:  "a,b",
+		},
+		{
+			name: "joined length is equal to limit",
+			elems: []string{
+				strings.Repeat("a", testLimit/2),
+				strings.Repeat("b", testLimit/2-testSepLen),
+			},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   strings.Repeat("a", testLimit/2) + "," + strings.Repeat("b", testLimit/2-testSepLen),
+		},
+		{
+			name: "joined length is greater than limit",
+			elems: []string{
+				strings.Repeat("a", testLimit/2),
+				strings.Repeat("b", testLimit/2),
+			},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   strings.Repeat("a", testLimit/2) + " (1 more omitted)",
+		},
+		{
+			name: "joined length is greater than limit and omission indicator doesn't fit",
+			elems: []string{
+				strings.Repeat("a", testLimit/2),
+				// This elem will be omitted as well as the next omission indicator wouldn't fit if we added it.
+				strings.Repeat("b", testLimit/2-testSepLen),
+				strings.Repeat("c", testLimit/2),
+			},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected:                   strings.Repeat("a", testLimit/2) + " (2 more omitted)",
+		},
+		{
+			name: "joined length is greater than limit and omission indicator fits perfectly",
+			elems: []string{
+				strings.Repeat("a", testLimit/2),
+				// This elem will be included because the next omission indicator fits just fine.
+				strings.Repeat("b", testLimit/2-len(" (1 more omitted)")-testSepLen),
+				strings.Repeat("c", testLimit/2),
+			},
+			separator:                  testSep,
+			limit:                      testLimit,
+			omissionIndicatorSeparator: testOmissionSep,
+			expected: strings.Repeat("a", testLimit/2) + "," +
+				strings.Repeat("b", testLimit/2-len(" (1 more omitted)")-testSepLen) + " (1 more omitted)",
```

Comment:
it's not clear what is the expected value. Instead, could you manipulate the limit and conditions to make sure expected value is not calculated and visible to the reader?

Reply:
Adjusted the test cases to use hardcoded expected values for readability, [PTAL](https://github.com/scylladb/scylla-operator/compare/a406b79b8d59ff2c55f96bc43946484d08e60652..8f457323b7c390f19295f3fd2446ca9f069cf0f5).